thumbor/engines/extensions/exif_orientation_editor.py

```diff
@@ -0,0 +1,66 @@
+import os
+import struct
+from io import BytesIO
+
+PREFIXES = [
+    b"MM\x00\x2A",  # Valid TIFF header with big-endian byte order
+    b"II\x2A\x00",  # Valid TIFF header with little-endian byte order
+    b"MM\x2A\x00",  # Invalid TIFF header, assume big-endian
+    b"II\x00\x2A",  # Invalid TIFF header, assume little-endian
+]
+
+EXIF_HEADER = b"Exif\x00\x00"
+
+
+class ExifOrientationEditor():
+    _endian = "<"
+    _offset = None
+
+    def __init__(self, data):
```

Comment:
Generally speaking, this `__init__` method is quite dense and very difficult to tell what is going on ... at least for a non-pythonist/non-image expert. It may be helpful for future devs to explain what is going on here via comments or even named functions.